runtime/appruntime/shared/health/health.go

```diff
@@ -0,0 +1,115 @@
+package health
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/exp/slices"
+)
+
+// CheckRegistry is a registry of health checks from the API and Infra SDKs
+// and other parts of the runtime.
+type CheckRegistry struct {
+	checks []Check
+	m      sync.Mutex
+}
+
+// NewCheckRegistry creates a new CheckRegistry.
+//
+// If running in an app there is a [Singleton]
+func NewCheckRegistry() *CheckRegistry {
+	return &CheckRegistry{}
+}
+
+// Register registers a new health check.
+//
+// Checks must complete within 5 seconds, otherwise
+// they will be terminated and considered failed.
+//
+// Checks can be called at any time and could have
+// multiple goroutines calling them concurrently.
+func (c *CheckRegistry) Register(check Check) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	c.checks = append(c.checks, check)
+}
+
+// RegisterFunc registers a new health check from a function with a given name
+//
+// Checks must complete within 5 seconds, otherwise
+// they will be terminated and considered failed.
+//
+// Checks can be called at any time and could have
+// multiple goroutines calling them concurrently.
```

Comment:
Nit: no need to duplicate all the documentation; better to refer to the other.

```suggestion
// RegisterFunc is like Register but constructs the Check from the given name and func.
```